bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/FlatLedgerManager.java

```diff
@@ -103,7 +103,8 @@ private synchronized void preload() throws IOException {
 
                 try {
                     zkActiveLedgers = ledgerListToSet(
-                            ZkUtils.getChildrenInSingleNode(zk, ledgerRootPath), ledgerRootPath);
+                            ZkUtils.getChildrenInSingleNode(zk, ledgerRootPath, ZkUtils.OP_TIME_OUT_SEC),
```

Comment:
What I mean is, it should probably be passed in from the gc thread.  The gc caller should probably get it's value from a config.  Doing it here doesn't allow different users with different tolerances to provide different bounds.  Also, passing in 0 should mean no limit.

Reply:
+1 to what @athanatos suggested. I would like to see a setting configurable.